### PR TITLE
Provider demo

### PR DESCRIPTION
**Pull request by tzhang-ext-aa**
please merge #2 first
try it:
`python -m pypact.consumer_test_main && pytest -sv pypact/demo/provider`

### CODE REVIEW

File: pypact/demo/provider/test_topchart_service.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class TopChartServiceTest(MockedTopChartService):
+    def test_topchart_service(self):
+        contract = contract_factory.register("TopChartService_Provider", "TopChartServiceTest_Consumer")
+        test_client = app.test_client()
+        for interaction in contract.interactions:
+            response = test_client.open(
+                interaction.request['path'],
+                method=interaction.request.setdefault("method", "GET"),
+            )
+            assert json.loads(response.data) == interaction.response['body']
```
Comment (luyun-aa) [inline pypact/demo/provider/test_topchart_service.py]: This doesn't not align with the sample code in the README. The "assert" part should not be implemented by service developers, but instead be implemented by the pypact library.

Pypact library is responsible to implement a specific version of [pact-specification](https://github.com/pact-foundation/pact-specification).

**Comment by tzhang-ext-aa** [comment]
moved to pytest, so please use `pytest -sv pypact/demo/provider` to run the provider test
@luyun-aa @chan-aa

File: pypact/demo/provider/test_topchart_service.py (hunk 1 above)
Comment (tzhang-ext-aa) [inline pypact/demo/provider/test_topchart_service.py]: fixed 🎉

File: pypact/demo/consumer/test_consumer_categroy.py [hunk 2]
```diff
@@ -16,7 +17,7 @@ def test_category_service(self):
                 headers={'Content=Type': 'application/json'},
                 body={"category": ["ALL", "GAMES", "APP"]}))
         category_service = CategoryService()
-
+        service = MockService(contract)
```
Comment (luyun-aa) [inline pypact/demo/consumer/test_consumer_categroy.py]: add `base_uri` as 2nd argument to MockService

**Pull request closed without merging**